### PR TITLE
Spike 462

### PR DESCRIPTION
**Pull request by Parth097**

### CODE REVIEW

File: src/app/components/navigation/TrackedRoute.tsx [hunk 1]
```diff
@@ -52,11 +53,16 @@ const WrapperComponent = function ({ component: Component, trackingOptions, ...p
   );
 };
 
+export const isGoogleBot = function (userAgent?: string): boolean {
+  const googleBotUserAgents: string = "compatible; Googlebot/2.1; +http://www.google.com/bot.html";
+  return userAgent !== undefined && userAgent.includes(googleBotUserAgents);
```
Comment (hannah-whelan) [inline src/app/components/navigation/TrackedRoute.tsx:58]: Sonarcloud suggests optional chaining here:

`return userAgent?.includes(googleBotUserAgents)`

File: src/test/services/googleBot.test.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    expect(isGoogleBot(validUserAgent)).toBe(true);
+  });
+
+  it("should return false for an invalid user agent", () => {
+    const invalidUserAgent = "Mozilla/5.0 (compatible; Bingbot/2.0; +http://www.bing.com/bot.html)";
+    expect(isGoogleBot(invalidUserAgent)).toBe(false);
+  });
+
+  it("should return false for undefined user agent", () => {
+    expect(isGoogleBot(undefined)).toBe(false);
```
Comment (hannah-whelan) [inline src/test/services/googleBot.test.ts:15]: undefined is unnecessary here, just call `isGoogleBot()`

File: src/app/components/navigation/TrackedRoute.tsx (hunk 1 above)
Comment (Parth097) [reply src/app/components/navigation/TrackedRoute.tsx:58]: I noticed that and already tried that method. 
However it is not working that way. There are no errors really, but as a user you can bypass login. 
That is why I kept it like that.

Comment (Parth097) [reply src/app/components/navigation/TrackedRoute.tsx:58]: This way is working:  

` return userAgent?.includes(googleBotUserAgents) ?? false;`

It was missing the last part

**Review by hannah-whelan — APPROVED** [review]

**Pull request merged**